### ISSUE TITLE
Link for outline requires both x and y coordinates in XYZ mode

### ISSUE THREAD

**Issue by King-of-Infinite-Space** [ArtifexSoftware/mupdf.js#89]
Hi devs, I want to add outlines to PDF that jump to specific y position of a page without affecting the x position.

Currently the outline won't work if only one of x, y is defined. It works if both x and y and defined.

```js
let outlineItem = {
  title: 'demo',
  uri: doc.formatLinkURI({ page: 0, type: 'XYZ', y: 200, }),
  open: false,
}

console.log(outlineItem)
// { title: 'demo', uri: '#page=1&zoom=nan,nan,200', open: false }
// this looks ok

doc.outlineIterator().insert(outlineItem)
console.log(doc.outlineIterator().item())
// { title: 'demo', uri: '#page=1', open: false }
// x,y coordinates are lost unless both are set
```

According to PDF spec (12.3.2.2 Explicit destinations), for /XYZ mode
>  A null value for *any* of the parameters left, top, or zoom specifies that the current value of that parameter shall be retained unchanged. 

Is this not implemented correctly or is there a workaround? Thanks.

**Comment by jamie-lemon** [comment]
@sebras I hope you can take a look at this one!

**Comment by sebras** [comment]
I have reported this as an [upstream bug](https://bugs.ghostscript.com/show_bug.cgi?id=707822). That one will be closed once it is fixed in the mupdf project, and this bug when the update reaches mupdf.js I assume.

**Comment by sebras** [comment]
The upstream bug has been fixed on upstream master. It is planned to be included in the upcoming 1.24.4 release within a few weeks.
